### PR TITLE
[ES] YPAY-100 upgrade to spartacus 2.0.0-next.4

### PR DESCRIPTION
**Pull request by e2yerik**

### CODE REVIEW

**Comment by declantyson** [comment]
CONFLICTS! :crossed_swords:

File: angular.json [hunk 1]
```diff
@@ -15,7 +15,12 @@
             "tsConfig": "projects/worldpay-spartacus/tsconfig.lib.json",
             "project": "projects/worldpay-spartacus/ng-package.json"
           }
-        },
+        ,          "configurations": {
```
Comment (declantyson) [inline angular.json:18]: can we reformat this file please

File: example-storefront/angular.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
             "main": "src/main.ts",
             "polyfills": "src/polyfills.ts",
             "tsConfig": "tsconfig.app.json",
-            "aot": false,
+            "aot": true,
```
Comment (declantyson) [inline example-storefront/angular.json:25]: What's this for?

File: angular.json (hunk 1 above)
Comment (Eusebiotrigo) [inline angular.json:18]: alignment?

File: example-storefront/src/app/app.module.ts [hunk 3]
```diff
@@ -49,7 +49,7 @@ import {
         fallbackLang: 'en'
       },
       features: {
-        level: '1.2'
+        level: '1.3'
```
Comment (declantyson) [inline example-storefront/src/app/app.module.ts:52]: Shouldn't this be 2.0?

File: example-storefront/tsconfig.app.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "types": []
   },
   "files": ["src/main.ts", "src/polyfills.ts"],
-  "include": ["src/**/*.ts"],
+  "include": ["src/**/*.d.ts"],
```
Comment (Eusebiotrigo) [inline example-storefront/tsconfig.app.json:8]: what's the difference here? d.ts?

File: projects/worldpay-spartacus/src/core/services/worldpay-checkout/worldpay-checkout.service.ts [hunk 5]
```diff
@@ -3,11 +3,12 @@ import { select, Store } from '@ngrx/store';
...
   CheckoutService,
   StateWithCheckout,
-  CartDataService
+  ActiveCartService,
+  AuthService
 } from '@spartacus/core';
 
 import * as WorldpayActions from '../../store/worldpay.action';
-import { Observable } from 'rxjs';
+import { combineLatest, Observable } from 'rxjs';
```
Comment (declantyson) [inline projects/worldpay-spartacus/src/core/services/worldpay-checkout/worldpay-checkout.service.ts:11]: Is this used?

File: projects/worldpay-spartacus/tsconfig.lib.prod.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.lib.json",
+  "angularCompilerOptions": {
+    "enableIvy": false
+  }
+}
```
Comment (Eusebiotrigo) [inline projects/worldpay-spartacus/tsconfig.lib.prod.json:6]: Activate "ensure line feed at the end of the file" in intellij.

File: angular.json (hunk 1 above)
Comment (e2yerik) [reply angular.json:18]: done

File: example-storefront/angular.json (hunk 2 above)
Comment (e2yerik) [reply example-storefront/angular.json:25]: this was part of the automated upgrade process: https://angular.io/guide/aot-compiler

File: example-storefront/src/app/app.module.ts (hunk 3 above)
Comment (e2yerik) [reply example-storefront/src/app/app.module.ts:52]: I can't see any documentation about the exact number for this. I think this version is more about the OCC version, not about the spartacus version. If you don't change this number to 1.3 the frontend will not work. will try with 2.0 instead

File: example-storefront/tsconfig.app.json (hunk 4 above)
Comment (e2yerik) [reply example-storefront/tsconfig.app.json:8]: this is part of the automated upgrade process.

File: projects/worldpay-spartacus/src/core/services/worldpay-checkout/worldpay-checkout.service.ts (hunk 5 above)
Comment (e2yerik) [reply projects/worldpay-spartacus/src/core/services/worldpay-checkout/worldpay-checkout.service.ts:11]: yes! line 34

File: projects/worldpay-spartacus/tsconfig.lib.prod.json (hunk 6 above)
Comment (e2yerik) [reply projects/worldpay-spartacus/tsconfig.lib.prod.json:6]: i have this enabled, see PM. This file is reformatted by the 'npm run format' command

File: projects/worldpay-spartacus/src/core/services/worldpay-checkout/worldpay-checkout.service.ts (hunk 5 above)
Comment (declantyson) [reply projects/worldpay-spartacus/src/core/services/worldpay-checkout/worldpay-checkout.service.ts:11]: i'm blind :sunglasses:

File: example-storefront/package.json [hunk 7]
```diff
@@ -15,51 +15,52 @@
...
+    "@angular/router": "~9.1.1",
+    "@angular/service-worker": "~9.1.1",
+    "@ng-bootstrap/ng-bootstrap": "^6.0.2",
+    "@ng-select/ng-select": "^4.0.0",
+    "@ngrx/effects": "^9.0.0",
+    "@ngrx/router-store": "^9.0.0",
+    "@ngrx/store": "^9.0.0",
     "@nguniversal/express-engine": "^8.1.1",
     "@nguniversal/module-map-ngfactory-loader": "8.1.1",
     "@spartacus/assets": "^1.1.0",
```
Comment (declantyson) [inline example-storefront/package.json:36]: You missed one!

Comment (declantyson) [reply example-storefront/package.json]: Fixed, also changed the config slightly to reference translations differently (it's changed at some point)

**Pull request merged**